### DISCUSSION TITLE
How to set transparency of a png and blur it using the CLI

### DISCUSSION THREAD

**Discussion by aacebedo** [libvips/libvips#3225] [Help wanted]
Hi 
I am trying to use the vips CLI to blur and set the transparency of a png image. I need a transparent background. With imagemagick I do it with the following command:
`convert img.in -blur 0x5 -matte -channel A -evaluate set 20% -channel rgba img.out`

with vips I blur it with this 
`vips gaussblur  int out 10`

But I don't know how to do the transparency part.

**Comment by jcupitt** [comment]
Hi @aacebedo,

You can change the transparency by multiplying by 0.2. FOr example:

```
vips gaussblur PNG_transparency_demonstration.png x.png 10
vips linear x.png y.png "1 1 1 0.2" 0
```

`linear in out a b` calculates `out = a * in + b`. You can use arrays for a and b and they'll represent image bands, so `1 1 1 0.2` means multiply RGB by 1 (ie. do nothing) and scale transparency by 0.2 (ie. reduce it to 20%).

Before:

![PNG_transparency_demonstration](https://user-images.githubusercontent.com/580843/207038561-5d111c64-998b-41b6-bbcf-8716745cb707.png)

After:

![y](https://user-images.githubusercontent.com/580843/207038607-77232261-2145-4534-be6b-72cd092686ab.png)

The libvips CLI is pretty basic -- it only lets you run a single operation each time. I would use something like python for this task, perhaps:

```python
#!/usr/bin/python3

import sys
import pyvips

image = pyvips.Image.new_from_file(sys.argv[1], access='sequential')
image = image.gaussblur(10) * [1, 1, 1, 0.2]
image.write_to_file(sys.argv[2])
```

**Comment by aacebedo** [comment]
Thanks for the answer. 
I use the CLI as it is only in a small script. Using python would be great but on my install it would require a lot to deal with deps.
I tried the linear command but it fails with `linear: vector must have 1 or 3 elements`

**Reply by jcupitt** [reply]
So it sounds like to have an RGB image, is that right? You'll need to add an alpha, perhaps:

```
vips gaussblur k2.jpg x.png 10
vips bandjoin_const x.png y.png 50
```

Adds an extra band with the constant value 50, which is about 255 * 0.2.

**Reply by aacebedo** [reply]
That's working ! the image was indeed in RGB.

Thank you
